persistence/test/module_test.go

```diff
@@ -1,35 +1,129 @@
 package test
 
 import (
-	"github.com/stretchr/testify/require"
 	"testing"
+
+	"github.com/pokt-network/pocket/shared/crypto"
+	"github.com/stretchr/testify/require"
 )
 
-func TestContextAndCommit(t *testing.T) {
+func TestContextAndCommitComplexFlow(t *testing.T) {
 	// variables for testing
 	poolName := "fake"
 	poolAddress := []byte("address")
 	originalAmount := "15"
 	modifiedAmount := "10"
-	// setup two separate contexts
-	contextA, err := testPersistenceModule.NewRWContext(0)
-	require.NoError(t, contextA.InsertPool(poolName, poolAddress, originalAmount))
-	require.NoError(t, contextA.Commit())
-	// verify the insert worked
-	contextA, err = testPersistenceModule.NewRWContext(0)
-	contextAOriginal, err := contextA.GetPoolAmount(poolName, 0)
+
+	// setup a context, insert a pool and commit it
+	context, err := testPersistenceModule.NewRWContext(0)
 	require.NoError(t, err)
-	require.Equal(t, originalAmount, contextAOriginal)
+	require.NoError(t, context.InsertPool(poolName, poolAddress, originalAmount))
+	require.NoError(t, context.Commit())
+
+	// verify the insert in the previously committed context worked
+	contextA, err := testPersistenceModule.NewRWContext(0)
 	require.NoError(t, err)
-	contextB, err := testPersistenceModule.NewRWContext(0)
+
+	contextAOriginalAmount, err := contextA.GetPoolAmount(poolName, 0)
 	require.NoError(t, err)
-	// modify only in context a and check that modification worked
+	require.Equal(t, originalAmount, contextAOriginalAmount)
+
+	// modify contextA but do not commit it
 	require.NoError(t, contextA.SetPoolAmount(poolName, modifiedAmount))
-	contextAAfter, err := contextA.GetPoolAmount(poolName, 0)
+
+	contextAModifiedAmount, err := contextA.GetPoolAmount(poolName, 0)
+	require.NoError(t, err)
+	require.Equal(t, modifiedAmount, contextAModifiedAmount)
+
+	// setup a second context - independent of the previous modified but uncommitted context
+	contextB, err := testPersistenceModule.NewReadContext(0)
+	require.NoError(t, err)
+
+	// verify context b is unchanged
+	contextBOriginalAmount, err := contextB.GetPoolAmount(poolName, 0)
+	require.NoError(t, err)
+	require.NotEqual(t, modifiedAmount, contextBOriginalAmount)
+	require.Equal(t, contextBOriginalAmount, contextAOriginalAmount)
+}
+
+// TODO(pocket/issues/149): Need to add support for this sort of test. The call to
+// `contextB.SetAccountAmount` currently hangs because we have multiple writes contexts at the
+// same height. Some potential solutions may include:
+// - keeping a set of all the write contexts and panicking
+// - Adding timeouts to the write contexts
+// - ???
+func TestTwoWriteContextsSameHeight(t *testing.T) {
```

Comment:
These tests don't make sense to me. As discussed, we should never have two write contexts in parallel

Reply:
Removed tests and am now erroring on multiple write contexts.